### ISSUE TITLE
host name not visible during cloud-init initialization and execuition of rc.local

### ISSUE THREAD

**Issue by Remigius2011** [rancher/os#2742]
**RancherOS Version: (ros os version)**

1.5.1 from https://releases.rancher.com/os/latest/rancheros-vmware.iso
provisioned through rancher 2.2.2 using a vSphere node template

**Where are you running RancherOS? (docker-machine, AWS, GCE, baremetal, etc.)**

on vSphere 6.7 - licensed

**My Problem:**

I am trying to set a static IP based on the node id (the number added to the name prefix. I have created a cloud-config script which is successfully executed during provisioning, but it does not see the ***real*** host name, only `rancher`. Of course, on the console the correct host name is displayed and contained in several locations where there is only `rancher` during provisioning (e.g. `/etc/hosts`, `machine-hostname.yml`).

Also when creating an `/etc/rc.local` script (which is executed successfully), I only see `rancher`, even if I delay the execution by invoking `wait-for-docker` followed by `sleep 5`.

How/when can I access some identity providing parameter (not necessarily the host name - but the host name would be perfect) during provisioning and change the network config based on it - if possible without interfering with k8s provisioning? Is there another decent way of individual node based provisioning with rancher / rancherOS / vSphere? When is `machine-hostname.yml`written?

I have searched for solutions, read the rancher OS documentation and tried numerous approaches to access the host name...

Of course, I have seen https://github.com/rancher/os/issues/2404 - but this does not solve my problem, firstly I don't (and don't want to) use DHCP, secondly I cannot use a static default value, as I want to handle a bunch of cluster nodes created and managed by rancher.

**Comment by Jason-ZW** [comment]
@Remigius2011  
The hostname default is your machine name. (https://github.com/docker/machine/blob/eae38c196a2e10e4e11cc7b8d6dfe8863fb60519/libmachine/provision/rancheros.go#L110). You can try to detect `/var/lib/rancher/conf/cloud-config.d/machine.yml` whether created or not. After created the `hostname` should be overwritten by docker-machine. I'm not sure whether this way can take effect.

**Comment by Remigius2011** [comment]
@Jason-ZW 
Hi Jason,

Thanks for helping.

The host name is - as I have meanwhile found out in experiments - unfortunately only set significantly later than the execution of the following:

* `rundmd` entries  in `cloud-config.yaml`
* a `cloud-config.sh` script
* `rc.local`

I have also attempted to start another script from `rc.local` that runs in the background and checks for the `hostname` to return something else than `rancher` - which not only worked eventually, but also allowed me to grab an IP address from a file and change the network configuration to use it - only to find out that this interferes with rancher provisioning (it looks like docker-machine is not aware of the changed IP address), so my whole idea of using static IP addreses (needed to bypass the squid proxy) did not work.

But - as often in life - sticking to an initial (wrong) idea will not succeed, but instead a whole new approach eventually got me a running cluster on vSphere a few minutes ago: I only had to configure the proxy server poroperly and the node was able to connect to the rancher server (bypassing the proxy) and to pull all the necessary images to install k8s (using the proxy).

My somewhat fixed idea was that cluster nodes will only work with direct internet access. But it looks like proper proxy (and no_proxy) settings will remediate this. I have set them now in `cloud-config.yaml` as well as in the docker environment in the node template (Engine Options > Engine Environment). I think however, that accessing the rancher server via a proxy still does not work due to the golang websocket client that lacks proxy support (at least this was the case some time ago).
